cmd/broscoli/main_test.go

```diff
@@ -0,0 +1,136 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"github.com/hypebeast/go-osc/osc"
+	"gopkg.in/yaml.v2"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strings"
+	"testing"
+	"time"
+)
+
+const (
+	SubCmdFlags = "SUB_CMD_FLAGS"
+)
+
+func TestAPI(t *testing.T) {
+	_ = os.Remove("/tmp/foo")
+	file, _ := os.Create("/tmp/bar")
+	defer file.Close()
+	tests := []struct {
+		name            string
+		actions         map[string]*Action
+		wantFileToExist bool
+		wantErr         bool
+	}{
+		{
+			name: "creating a file",
+			actions: map[string]*Action{"/create": {
+				Command:    "/bin/touch /tmp/foo",
+				Wait:       false,
+				execParams: nil,
+			},
+			},
+			wantFileToExist: true,
+			wantErr:         false,
+		},
+		{
+			name: "removing a file",
+			actions: map[string]*Action{"/remove": {
+				Command:    "/bin/rm /tmp/bar",
+				Wait:       false,
+				execParams: nil,
+			},
+			},
+			wantFileToExist: false,
+			wantErr:         false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			address := "0.0.0.0"
+			port := uint16(8765)
```

Comment:
This also prevents concurrent tests from running. :thinking:

Reply:
We might also just rework the code and move most of it to a package of its own, so that `main()` is only left with config file parsing and a call to `.Run()` into that package. Then the package could have the tests.

Reply:
Ideally yes, but then at this time there's not that much code at this moment to go overboard.